backend/src/ee/services/permission/permission-types.ts

```diff
@@ -1,9 +1,41 @@
-export type TBuildProjectPermissionDTO = {
-  permissions?: unknown;
-  role: string;
-}[];
+import picomatch from "picomatch";
+import { z } from "zod";
 
-export type TBuildOrgPermissionDTO = {
-  permissions?: unknown;
-  role: string;
-}[];
+export enum PermissionConditionOperators {
+  $IN = "$in",
+  $ALL = "$all",
+  $REGEX = "$regex",
+  $EQ = "$eq",
+  $NEQ = "$ne",
+  $GLOB = "$glob"
+}
+
+export const PermissionConditionSchema = {
+  [PermissionConditionOperators.$IN]: z.string().array(),
+  [PermissionConditionOperators.$ALL]: z.string().array(),
+  [PermissionConditionOperators.$REGEX]: z.string().refine(
+    (el) => {
+      try {
+        // eslint-disable-next-line no-new
+        new RegExp(el);
+        return true;
+      } catch {
+        return false;
+      }
+    },
+    { message: "Invalid regex pattern" }
+  ),
```

Comment:
should we add a check for computationally expensive regex rules? I know some regex patterns can easily bring down a server

Reply:
We don't use regex. This is for phase 2 and regex won't be used against database. it will be like a filter ran after the db data.